core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -4799,6 +4812,99 @@ && isHourKeyword(qc.getAst().token) && qc.getAst().paramCount == 1 && qc.getAst(
         return root;
     }
 
+
+    private boolean checkIfTranslatingModelIsRedundant(boolean useInnerModel, boolean useGroupByModel, boolean useWindowModel, boolean forceTranslatingModel, boolean checkTranslatingModel, QueryModel translatingModel) {
+        // check if translating model is redundant, e.g.
+        // that it neither chooses between tables nor renames columns
+        boolean translationIsRedundant = (useInnerModel || useGroupByModel || useWindowModel) && !forceTranslatingModel;
+        if (translationIsRedundant && checkTranslatingModel) {
+            for (int i = 0, n = translatingModel.getBottomUpColumns().size(); i < n; i++) {
+                QueryColumn column = translatingModel.getBottomUpColumns().getQuick(i);
+                if (!column.getAst().token.equals(column.getAlias())) {
```

Comment:
we should be using `Chars.equals()` here, Java doesn't compare String to CharSequences well. Also comparison should be case insensitive (+test that proves that)

Reply:
> we should be using `Chars.equals()` here, Java doesn't compare String to CharSequences well. Also comparison should be case insensitive (+test that proves that)

Good catch!

@nwoolmer we have lots of `CharSequence` implementations here and there and that's why we have these utility methods to be able to compare them with each other and `java.lang.String`.